src/plugins/embeddable/server/plugin.ts

```diff
@@ -129,6 +131,29 @@ export class EmbeddableServerPlugin implements Plugin<object, object> {
     return updatedInput;
   };
 
+  private migrate = (state: EmbeddableStateWithType, version: string) => {
```

Comment:
Comparing `server/plugin.ts` and `public/plugin.tsx`, it seems the logic of `migrate` is basically identical, with a few small semantic differences. (there's a lot of duplication elsewhere too, like telemetry, extract, inject)

Then there's this `baseEmbeddableMigrations` which is imported from a `common` file exclusively for use here.

I wonder if it would reduce the possibility of future errors/divergence if we colocated `migrate` with the other `migrate_base_input` stuff and then imported that into both client & server plugin.ts files

Reply:
This would also have the benefit of ensuring we have test coverage -- looks like the client-side `plugin.tsx` has unit tests but not `server/plugin.ts`?